velox/docs/functions/spark/array.rst

```diff
@@ -51,6 +51,18 @@ Array Functions
         SELECT array_min(ARRAY [4.0, float('nan')]); -- 4.0
         SELECT array_min(ARRAY [NULL, float('nan')]); -- NaN
 
+.. function:: array_remove(x, element) -> array
+
+    Remove all elements that equal ``element`` from array ``x``. Returns NULL as result if ``element`` is NULL.
+    If array ``x`` is empty array, returns empty array. If all elements in array ``x`` are NULL but ``element`` is not NULL,
+    returns array ``x``. ::
```

Comment:
@mbasmanova I just remembered the results of these array functions (array_union, array_except etc.) on NaN are different between Presto and Spark. According to https://github.com/facebookincubator/velox/pull/7237, will NaNs be treated as equal in these functions? If so, the semantics will be aligned with Spark. Otherwise, we may need a separate implementation for Spark.

```
> scala> spark.sql("SELECT array_remove(ARRAY(2, 1, cast('NaN' as double)), cast('NaN' as double))").show()
+-------------------------------------------------------------------+           
|array_remove(array(2, 1, CAST(NaN AS DOUBLE)), CAST(NaN AS DOUBLE))|
+-------------------------------------------------------------------+
|                                                         [2.0, 1.0]|
+-------------------------------------------------------------------+
```
Presto 
```
presto:test> SELECT array_remove(ARRAY [2, 1, cast('NaN' as double)], cast('NaN' as double)); 
      _col0      
-----------------
 [2.0, 1.0, NaN] 
(1 row)
```

Reply:
I'm working on `collect_set` agg func for Spark, I found that NaNs will be treated as not equal in Spark's `collect_set` which is semantic equivalent to Presto's `set_agg`.
Spark
```
spark-sql> SELECT collect_set(col) FROM VALUES (1.0), (2.0), (cast('NaN' as double)), (cast('NaN' as double)) AS tab(col);
[2.0,1.0,NaN,NaN]
```
Presto
```
presto:default> SELECT set_agg(col) FROM (VALUES (1.0), (2.0), (cast('NaN' as double)), (cast('NaN' as double))) AS tab(col);
        _col0
----------------------
 [1.0, 2.0, NaN, NaN]
```